processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -85,13 +87,13 @@
 
     private final List<String> enumConstants;
 
-    private Map<String, ExecutableElement> getters = null;
-    private Map<String, ExecutableElement> presenceCheckers = null;
+    private Map<String, Accessor> readAccessors = null;
+    private Map<String, ExecutableElementAccessor> presenceCheckers = null;
 
-    private List<ExecutableElement> allExecutables = null;
-    private List<ExecutableElement> setters = null;
-    private List<ExecutableElement> adders = null;
-    private List<ExecutableElement> alternativeTargetAccessors = null;
+    private List<Accessor> allAccessors = null;
+    private List<Accessor> setters = null;
+    private List<Accessor> adders = null;
```

Comment:
Yes they will be, they won't be fields. They are filtered from the `allAccessors`. I was thinking that it is better if we have something more generic, and not have the specific implementations referenced in other places